sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -195,9 +202,28 @@ case class Aggregate(
     child: LogicalPlan)
   extends UnaryNode {
 
+  override lazy val resolved: Boolean = {
+    val hasWindowExpressions = aggregateExpressions.exists ( _.collect {
+        case window: WindowExpression => window
+      }.nonEmpty
+    )
+
+    !expressions.exists(!_.resolved) && childrenResolved && !hasWindowExpressions
+  }
+
   override def output: Seq[Attribute] = aggregateExpressions.map(_.toAttribute)
 }
 
+case class Window(
+    projectList: Seq[Attribute],
+    windowExpressions: Seq[NamedExpression],
+    windowSpec: WindowSpecDefinition,
```

Comment:
Since a `Window` operator evaluate a single `WindowSpecDefinition`, it is just easy to keep it at here instead of search it from the expressions.